examples/campaign_management/set_ad_parameters.py

```diff
@@ -0,0 +1,114 @@
+    operations.append(create_ad_parameter(client, resource_name, 1, '100'))
+    operations.append(create_ad_parameter(client, resource_name, 2, '$40'))
+
+    ad_parameter_service = client.get_service('AdParameterService',
+                                              version='v2')
+
+    # Add the ad parameter.
+    try:
+        response = ad_parameter_service.mutate_ad_parameters(customer_id,
+                                                             operations)
+    except GoogleAdsException as ex:
+        print('Request with ID "%s" failed with status "%s" and includes the '
+              'following errors:' % (ex.request_id, ex.error.code().name))
+        for error in ex.failure.errors:
+            print('\tError with message "%s".' % error.message)
+            if error.location:
+                for field_path_element in error.location.field_path_elements:
+                    print('\t\tOn field: %s' % field_path_element.field_name)
+        sys.exit(1)
+    else:
+        print('Set {} ad parameters:'.format(len(response.results)))
+
+        for result in response.results:
+            print('Set ad parameter with resource_name: {}'.format(
+                result.resource_name))
+
+def create_ad_parameter(client, resource_name, parameter_index, insertion_text):
+    """Creates a new ad parameter create operation and returns it.
+
+    There can be a maximum of two ad parameters per ad group criterion, one
+    with "parameter_index" = 1 and another with "parameter_index" = 2.
+
+    Restrictions apply to the value of the "insertion_text". For more
+    information, see the field documentation in the AdParameter class located
+    here: https://developers.google.com/google-ads/api/fields/v2/ad_parameter#ad_parameterinsertion_text
+
+    Args:
+        client: An initialized GoogleAdsClient instance.
+        resource_name: The resource name of the ad group criterion.
+        parameter_index: The unique index int of this ad parameter..
+        insertion_text: A numeric str value to insert into the ad text.
+
+    Returns: A new AdParamterOperation message class instance.
+    """
+    ad_param_operation = client.get_type('AdParameterOperation', version='v2')
+    ad_param = ad_param_operation.create
+    ad_param.ad_group_criterion.value = resource_name
+    ad_param.parameter_index.value = parameter_index
+    ad_param.insertion_text.value = insertion_text
+    return ad_param_operation
+
+
+if __name__ == '__main__':
+    # GoogleAdsClient will read the google-ads.yaml configuration file in the
+    # home directory if none is specified.
+    google_ads_client = GoogleAdsClient.load_from_storage()
+
+    # Initializes a command line argument parser.
+    parser = argparse.ArgumentParser(
+        description='Adds an ad group for specified customer and campaign id.')
+
+    # The following argument(s) should be provided to run the example.
+    parser.add_argument('-c', '--customer_id', type=six.text_type,
```

Comment:
These arguments are longs in Java. Why not set `type=six.integer_types` so validation is done on the command line?

Reply:
`add_argument` expects the `type` param to callable, and `six.integer_types` is actually a tuple, (to support `long` and `int` in Python 2) so using it generates an error.

We could set the type to `int` (since there is no `long` type in Python 3). That would generate an error immediately if someone tried to use a non-numeric value for a customer ID. The problem then is that we'd need to convert the numeric value to an instance of `str` or `bytes` before passing them into the service methods, otherwise they'll throw an error.

Actually it looks like they do [exactly that](https://github.com/googleads/google-ads-java/blob/master/google-ads-examples/src/main/java/com/google/ads/googleads/examples/basicoperations/GetCampaigns.java#L92) in Java so I'm not sure why they require the CIDs to be Longs in the examples.

In any case I'd prefer to stick with strings for these since that's what the service methods require.

Reply:
ok, let's stick with strings and not concern about consistency of type safeness right now.